### DISCUSSION TITLE
gitlog_showcase.rmd not properly checking out logs?

### DISCUSSION THREAD

**Discussion by Ssunoo2** [sailuh/kaiaulu#269] [Q&A] [answered]
Line 88 ( git_checkout(git_branch,git_repo_path) ) returns this error. 

Warning: running command ''git' --git-dir ../../rawdata/git_repo/APR/.git --work-tree ../../rawdata/git_repo/APR/ checkout trunk 2>/dev/null' had status 128character(0)
attr(,"status")
[1] 128

This causes line 89 ( project_git <- parse_gitlog(perceval_path,git_repo_path)  )to return this error: 
 Show Traceback
Error in parse_gitlog(perceval_path, git_repo_path) : 
  Unable to generate git log from this repository. Perhaps the path specified was incorrect or the repository has no commits?

I’m assuming that because 128 indicates triggering an exit command without a valid exit code that something went wrong with this command and the git logs were not downloaded but I’m not sure what could have gone wrong. The second error is simply explaining that the path may be wrong, indicating that the .git file is not present in the pathway ( ../../rawdata/git_repo/APR/.git ), which I'm assuming is a symptom of the first error. 

I double checked that the pathway was correct (though I'm unsure if I need to create an empty .git file or if that will be created automatically). The branch parameter has two metrics: - 1.2.9 and -trunk. Using the link on the config file (https://github.com/apache/apr) , the branch is named trunk, though I'm not sure where the 1.2.9 is coming from and if this may be causing the issue.

**Comment by carlosparadis** [comment, marked as answer]
@Ssunoo2 possibly duplicate to https://github.com/sailuh/kaiaulu/discussions/265? And also potentially related to confusion on using relative paths as in https://github.com/sailuh/kaiaulu/discussions/264? 

Have you downloaded the project's git log? You have to place it on the path you specify on APR project configuration file. Have a look on the narrative on the Notebook, it should explain what a project configuration file is, and what each path specifying there should be.

Also, as asked on the jira download comment, let me know your folder structure where `kaiaulu-master` is. It maybe a symptom of looking for files where it does not exist. This pushes up the need to improve clarity on these error messages with better advice.
